### PR TITLE
HPCC-31379 Fix new OSX 13 compiler warnings

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: ecl/hql/CMakeLists.txt [hunk 1]
```diff
@@ -167,6 +167,9 @@ if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
 
   #Compiler complains about an unused function yyunput, and make sure that any warnings are not reported as errors
   set_source_files_properties (hqllex.cpp PROPERTIES COMPILE_FLAGS " -Wno-unused-function -Wno-error")
+
+  #Compiler complains about a variable is set but not used 
+  set_source_files_properties (hqlgram.cpp PROPERTIES COMPILE_FLAGS " -Wunused-but-set-variable -Wno-error")
```
Comment (ghalliday) [inline ecl/hql/CMakeLists.txt:172]: I think this should be
```
set_source_files_properties (hqlgram.cpp PROPERTIES COMPILE_FLAGS " -Wno-unused-but-set-variable")
```

**Review by ghalliday — COMMENTED** [review]
One minor question/comment.
Also, should this target 9.4, in case we need to build that with the new compiler in the future?

**Comment by GordonSmith** [comment]
> One minor question/comment. Also, should this target 9.4, in case we need to build that with the new compiler in the future?

That will only happen if GH-Actions drop support for OSX-12 and we can backport then?  (The same argument is true of Ubuntu-24.04 and 22.04).

File: ecl/hql/CMakeLists.txt (hunk 1 above)
Comment (GordonSmith) [reply ecl/hql/CMakeLists.txt]: Ended up changing to:  `-Wno-error=unused-but-set-variable -Wno-error=class-memaccess`

**Comment by ghalliday** [comment]
> > One minor question/comment. Also, should this target 9.4, in case we need to build that with the new compiler in the future?
> 
> That will only happen if GH-Actions drop support for OSX-12 and we can backport then? (The same argument is true of Ubuntu-24.04 and 22.04).

@GordonSmith 

...or if a developer on a modern mac want's to build an older version of the platform.  That has bitten me a few times in the past.  For that reason I tend to target safe compiler-related fixes to older versions.  Please can you see if it cleanly rebases back to 9.4.x.  If so I'll merge it there, otherwise to master.

**Comment by ghalliday** [comment]
@GordonSmith please can you rebase onto 9.4.x.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**